### DISCUSSION TITLE
Python runtime for Flutter apps

### DISCUSSION THREAD

**Discussion by jet10000** [nylo-core/nylo#68] [Ideas]
https://github.com/flet-dev/serious-python

Integrate this component into Nylo, it can run python server in fultter app, thanks.

**Comment by agordn52** [comment]
Hi @jet10000,

Thanks for the idea but I have to ask, why would this be useful to a Flutter developer? 
What things would it allow us to build better in Flutter?

Thanks

**Comment by jet10000** [comment]
For example, you can let python process pictures, do some complex calculations, calculations in specific tasks, and build a background web server in client-side.

Flutter app do more than read and write data, such as client-side machine learning.

Let flutter focus on UI.

**Reply by agordn52** [reply]
It does sound interesting! Thanks for sharing but I'm trying to think what 'complex calculations' it could actually help with.
Since most projects built with Flutter are mobile projects (usually just making API requests), the data would always be handled by the server.

Re. ML
I can't answer acutely as I don't have experience using ML with my projects.
ML will probably play a bigger role in the future for Flutter but at the moment, I'm not sure what things it could actually help with unless the developer is building something that specifically requires it. 

If they do need ML, at the moment there are a few packages on pub.dev [here](https://pub.dev/packages?q=Machine+Learning).

**Comment by jet10000** [comment]
It can be used for computing, for storage, for page sharing data, for caching, for proxying requests, and so on. 

It can host local http services and websockets services. Also think of it as a role between the flutter app and the backend server. 

No need to write a lot of complex calculation logic in dart, and the python ecosystem is very rich.

let flutter focus on UI, and let it do complex client logic.

**Comment by dhsont** [comment]
I think it is overkill for micro framework,  we should keep it simple.

**Reply by andim27** [reply]
I think so - no need to overload micro framework.The better choice is develop independent plugin for ML ability
